src/Database/Type/DateTimeType.php

```diff
@@ -172,9 +172,13 @@ public function marshal($value)
             $value['minute'],
             $value['second']
         );
-        $tz = isset($value['timezone']) ? $value['timezone'] : null;
 
-        return new $class($format, $tz);
+        $timezone = null;
+        if (isset($value['localization'])) {
+            list($locale, $timezone) = explode(':', $value['localization']);
```

Comment:
What happens when there is no colon? At this point $value is user data, so we have to expect and assume that the data will be wrong/corrupted.